File: examples/dotnet/Calculator.cs
namespace Calculator;

public class Calc
{
    public static int Add(int a, int b) => a + b;
    public static int Subtract(int a, int b) => a - b;
    public static int Multiply(int a, int b) => a * b;
    public static double Divide(int a, int b)
    {
        if (b == 0) throw new DivideByZeroException();
        return (double)a / b;
    }
}

public class Program
{
    public static void Main()
    {
        Console.WriteLine($"2 + 3 = {Calc.Add(2, 3)}");
        Console.WriteLine($"5 - 2 = {Calc.Subtract(5, 2)}");
        Console.WriteLine($"4 * 3 = {Calc.Multiply(4, 3)}");
        Console.WriteLine($"10 / 2 = {Calc.Divide(10, 2)}");
    }
}

File: examples/java/src/Calculator.java
public class Calculator {
    public static int add(int a, int b) {
        return a + b;
    }

    public static int subtract(int a, int b) {
        return a - b;
    }

    public static int multiply(int a, int b) {
        return a * b;
    }

    public static double divide(int a, int b) {
        if (b == 0) {
            throw new ArithmeticException("Cannot divide by zero");
        }
        return (double) a / b;
    }

    public static void main(String[] args) {
        System.out.println("2 + 3 = " + add(2, 3));
        System.out.println("5 - 2 = " + subtract(5, 2));
        System.out.println("4 * 3 = " + multiply(4, 3));
        System.out.println("10 / 2 = " + divide(10, 2));
    }
}
